imcsdk/apis/server/remotepresence.py

```diff
@@ -181,6 +212,106 @@ def vmedia_mount_add(handle, volume_name,
     return vmediamap_mo
 
 
+def vmedia_mount_iso_uri(handle, uri, user_id=None, password=None,
+                         timeout=60, interval=5):
+    """
+    This method will setup the vmedia mapping
+    Args:
+        handle (ImcHandle)
+        uri (string): URI of the ISO image
+        user_id (string): optional username
+        password (string): optional password
+        timeout (int): optional timeout to wait for ISO map status to be 'OK'
+        interval (int): optional interval to query ISO status
+
+    Raises:
+        Exception if invalid protocol in URI
+        Exception when the mapping doesn't reach 'OK' status
+
+    Returns:
+        True if mapping succeeded
+
+    Examples:
+        vmedia_mount_add(
+            handle,
+            uri="http://10.127.150.13/files/ubuntu-14.04.2-server-amd64.iso"
+        )
+    """
+
+    # Verify interval not set to zero
+    if interval < 1 or type(interval) is not int:
+        raise ValueError("ERROR: interval must be positive integer")
+
+    # Parse file/path from URI
+    remote_file = os.path.basename(uri)
+    remote_share = os.path.dirname(uri) + "/"
+    mount_options = "noauto"
+
+    # Set the Map based on the protocol
+    if urlparse.urlsplit(uri).scheme == 'http':
+        mount_protocol = "www"
+    elif urlparse.urlsplit(uri).scheme == 'https':
+        mount_protocol = "www"
+    elif CIFS_URI_PATTERN.match(uri):
+        mount_protocol = "cifs"
+    elif NFS_URI_PATTERN.match(uri):
+        mount_protocol = "nfs"
+    else:
+        # Raise ValueError and bail
+        raise ValueError("Unsupported protocol: " +
+                         urlparse.urlsplit(uri).scheme)
+
+    # Convert no user/pass to blank strings
+    if not user_id:
+        user_id = ''
+    if not password:
+        password = ''
```

Comment:
@britthouser I don't think this was added by you, by a cleaner way to write the above 4 line is,

```
user_id = user_id or ''
password = password or ''
```